packages/kbn-i18n/src/core/i18n.js

```diff
@@ -73,6 +73,12 @@ export function addMessages(newMessages = {}, locale = newMessages.locale) {
     throw new Error('[I18n] A `locale` must be a non-empty string to add messages.');
   }
 
+  if (newMessages.locale && newMessages.locale !== locale) {
+    throw new Error(
+      '[I18n] A `locale` in the messages object is different from one provided as second argument.'
```

Comment:
__typo:__ not a native speaker, but something like `from the one provided as a second argument` sounds a little bit better, up to you :man_shrugging: